data_structures/src/transaction_factory.rs

```diff
@@ -263,6 +264,13 @@ pub fn get_total_balance(
         })
         .sum();
 
+    if simple {
+        return NodeBalance {
+            confirmed: new_balance,
```

Comment:
Instead of duplicating the total balance, the `confirmed` field should be an `Option<u64>`, set to `None` if the confirmed balance is not known.

```suggestion
            confirmed: None,
```